test/functional/tests/write_back/test_wb_acp.py

```diff
@@ -0,0 +1,171 @@
+            if cache.get_dirty_blocks() == initial_dirty_blocks:
+                TestRun.fail(f"No data flushed in 10s.\n{str(cache.get_statistics())}")
+            initial_dirty_blocks = cache.get_dirty_blocks()
+
+        TestRun.LOGGER.info(str(cache.get_statistics()))
+
+        action_kind = ActionKind.IoHandled
+        output = trace.stop_monitoring()
+        blktrace_output = [h for h in output if h.action == action_kind
+                           and RwbsKind.F not in h.rwbs]
+
+        if not blktrace_output:
+            TestRun.fail(f"No {action_kind.name} entries in blktrace output!")
+        else:
+            TestRun.LOGGER.debug(f"Blktrace headers count: {len(blktrace_output)}.")
+
+    with TestRun.step("Using blktrace verify that cleaning thread cleans data from "
+                      "all CAS device parts in proper order."):
+        write_counter = 0
+        all_writes_ok = True
+        no_chunk_left = False
+        last_sector = None
+        max_percent = 100
+        bucket_chunks = []
+        current_chunk = None
+
+        for header in blktrace_output:
+            if no_chunk_left:
+                TestRun.LOGGER.error(f"No chunk left for sector {header.sector_number} "
+                                     f"({str(Size(header.sector_number, Unit.Blocks512))}).")
+                continue
+
+            # Sector not in current chunk - search for the next chunk
+            if current_chunk is None or not sector_in_chunk(current_chunk[0], header.sector_number):
+                # Search for bucket with chunks that contain most dirty data
+                while not bucket_chunks and max_percent > 0:
+                    bucket_chunks = [a for a in chunk_list
+                                     if max_percent >= a[1] > max_percent - 10]
+                    max_percent -= 10
+
+                if not bucket_chunks:
+                    TestRun.LOGGER.error(f"No chunks left for sector {header.sector_number} "
+                                         f"({Size(header.sector_number, Unit.Blocks512)}).")
+
+                # Get chunk within current bucket where current header sector is expected
+                chunk = next((c for c in bucket_chunks if
+                              sector_in_chunk(c[0], header.sector_number)), None)
+                if not chunk:
+                    TestRun.LOGGER.error(f"Sector {header.sector_number} "
+                                         f"({Size(header.sector_number, Unit.Blocks512)}) "
+                                         f"not in current bucket.")
+                    all_writes_ok = False
+                    if all(lambda s: not sector_in_chunk(s[0], header.sector_number, chunk_list)):
+                        TestRun.LOGGER.warning(f"Sector {header.sector_number} "
+                                               f"({Size(header.sector_number, Unit.Blocks512)}) "
+                                               f"outside of any tested chunk.")
```

Comment:
Hmm, you're first logging an error for chunk not in current bucket and then just warning if it's not in all chunks, and I wonder if it's the right way to go. Do we want just warning for sectors written outside of all chunks (if that happens then it's not an ACP write) or do we treat it as a proper error (do we expect and allow any writes besides those generated by ACP?)